dashboard/app/controllers/api_controller.rb

```diff
@@ -206,7 +206,7 @@ def user_progress_for_stage
         script_level_id: script_level.try(:id),
         level_id: level.id,
         user_agent: request.user_agent,
```

Comment:
Should we add the same defensive fix here, if it can contain invalid characters?